File: nodes/dataset/create-dataset/create-dataset.js
module.exports = function(RED){
	function createDatasetNode(config){
		const path = require('path')
		const utils = require('../../../utils/utils')

		var node = this;

		//set configurations
		node.file = __dirname + '/create-dataset.py'
		node.config = {
			path: config.path,
			save: path.join(config.saveFolder, config.saveName),
			input: utils.listOfInt(config.input) || [0],
			output: parseInt(config.output) || undefined,
			trainingPartition: (Number(config.trainingPartition) || 80.0) / 100.0,
			shuffle: Boolean(config.shuffle),
			seed: parseInt(config.seed) || 0
		}

		utils.run(RED, node, config)
	}
	RED.nodes.registerType("create dataset", createDatasetNode)
}

File: nodes/dataset/create-dataset/create-dataset.py
import json
import pandas
import os

#read configurations
config = json.loads(input())

while True:
	#wait request
	input()

	df = pandas.read_csv(config['path'], header=None)

	r, c = df.shape

	if config['shuffle']:
		df = df.sample(frac=1, random_state=config['seed'])

	x_train = df.iloc[: int(r * config['trainingPartition']), config['input']]
	x_test = df.iloc[int(r * config['trainingPartition']) :, config['input']]

	if 'output' in config:
		y_train = df.iloc[: int(r * config['trainingPartition']), config['output']]
		y_test = df.iloc[int(r * config['trainingPartition']) :, config['output']]

		x_train[x_train.shape[1]] = y_train
		x_test[x_test.shape[1]] = y_test

	if not os.path.isdir(config['save']):
		os.makedirs(config['save'], exist_ok=True)

	x_train.to_csv(config['save'] + '/train.csv', header=None, index=False)
	x_test.to_csv(config['save'] + '/test.csv', header=None, index=False)

	print('Dataset created.')
